### PR TITLE
Refactoring and fixing PayableToSupplierOrHotel calculation

### PR DESCRIPTION
**Pull request by konstyefremov**
https://github.com/happy-travel/agent-app-project/issues/436#issuecomment-889117211

### CODE REVIEW

File: Api/Services/Reports/ReportService.cs [hunk 1]
```diff
@@ -74,12 +76,20 @@ public async Task<Result<Stream>> GetSalesBookingsReport(DateTime fromDate, Date
...
+                .Ensure(NoFutureDate, "Cannot query future dates for this report")
                 .Map(GetRecords)
                 .Bind(Generate<SalesBookingsReportData, SalesBookingsReportRow>);
 
 
             Task<IEnumerable<SalesBookingsReportData>> GetRecords()
                 => GetRecords<SalesBookingsReportData>(from, to);
+
+
+            bool NoFutureDate()
```
Comment (mingazhev) [inline Api/Services/Reports/ReportService.cs:88]: DatesAreNotInFuture

Comment (mingazhev) [reply Api/Services/Reports/ReportService.cs:88]: + this can be included into "Validate" mtehod

**Review by mingazhev — APPROVED** [review]
With small comments

**Review by mingazhev — CHANGES_REQUESTED** [review]
Need to optimize the query not to load all the bookings

File: Api/Services/Reports/ReportService.cs (hunk 1 above)
Comment (konstyefremov) [reply Api/Services/Reports/ReportService.cs]: Done

**Review by mingazhev — APPROVED** [review]

**Pull request merged**